channel/handlers.go

```diff
@@ -2979,3 +2993,167 @@ func (server Server) handleChatEvent(conn mnet.Server, reader mpacket.Reader) {
 		log.Println("Unknown chat event type:", op)
 	}
 }
+
+func (d *player) removeAllCooldowns() {
+	if d == nil || d.skills == nil {
+		return
+	}
+	for id, ps := range d.skills {
+		ps.Cooldown = 0
+		ps.TimeLastUsed = 0
+		d.updateSkill(ps)
+		_ = id
+	}
+}
+
+func (server *Server) getAffectedPartyMembers(p *party, affectedMask byte) []*player {
```

Comment:
Should you not be taking map and instance into account for this? Party members on the same map in the same instance should be affected